### PR TITLE
feat: Add support for Hive 4.0.0

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

Part of https://github.com/stackabletech/issues/issues/620
Needs https://github.com/stackabletech/docker-images/pull/818

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [x] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: tests/test-definition.yaml [hunk 1]
```diff
@@ -19,13 +19,13 @@ dimensions:
       - 3.1.3
       # Alternatively, if you want to use a custom image, append a comma and the full image name to the product version
       # as in the example below.
-      # - 3.1.3,docker.stackable.tech/sandbox/hive:3.1.3-stackable0.0.0-dev
+      - 4.0.0,docker.stackable.tech/sandbox/hive:4.0.0-stackable0.0.0-dev
```
Comment (maltesander) [inline tests/test-definition.yaml:22]: TODO: switch to stackable image once built

Comment (sbernauer) [reply tests/test-definition.yaml]: Done, waiting for https://github.com/stackabletech/docker-images/actions/runs/10680049849

File: tests/test-definition.yaml [hunk 2]
```diff
@@ -17,15 +17,16 @@ dimensions:
...
       - 3.1.3
+      - 4.0.0
       # Alternatively, if you want to use a custom image, append a comma and the full image name to the product version
       # as in the example below.
-      # - 3.1.3,docker.stackable.tech/sandbox/hive:3.1.3-stackable0.0.0-dev
+      # - 4.0.0,docker.stackable.tech/sandbox/hive:4.0.0-stackable0.0.0-dev
   - name: hive-latest
     values:
-      - 3.1.3
+      - 4.0.0
```
Comment (maltesander) [inline tests/test-definition.yaml:26]: We had this plan with replacing that "latest" thing. I would rather have some tests for the experimental and have all tests run on the LTS version?
So i would argue to put the 3.1.3 here rather 4.0.0 even if it does not make sense semantically?

**Review by maltesander — APPROVED** [review]
LGTM!

**Comment by sbernauer** [comment]
Started https://testing.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/hive-operator-it-custom/2/

**Comment by sbernauer** [comment]
Tests passed :rocket:

**Pull request merged**